src/frontend/sharedComponents/SelectDevice.tsx

```diff
@@ -0,0 +1,226 @@
+
+  return (
+    <ScrollView style={styles.container} testID="PROJECT.invite-device-scrn">
+      <View style={{flexDirection: 'row', alignItems: 'center'}}>
+        <WifiIcon style={{marginRight: 10}} width={30} height={30} />
+        <BodyText>{ssid}</BodyText>
+      </View>
+      <HeaderText variant="header6" style={{marginTop: 10}}>
+        {t(m.notSeeingDevice)}
+      </HeaderText>
+      <BodyText style={{marginLeft: 10}}>{`\u2022 ${t(m.sameWifi)}`}</BodyText>
+      <BodyText
+        style={{marginLeft: 10}}>{`\u2022 ${t(m.sameVersion)}`}</BodyText>
+      {/* Divider */}
+      <View style={{marginTop: 20}} />
+
+      <View style={styles.deviceListContainer}>
+        {selectableDevices.map(device => {
+          const {deviceId, status, name, deviceType} = device;
+
+          let isDisconnected: boolean;
+          switch (status) {
+            case undefined:
+            case 'connected':
+              isDisconnected = false;
+              break;
+            case 'disconnected':
+              isDisconnected = true;
+              break;
+            default:
+              throw new ExhaustivenessError(status);
+          }
+
+          const handlePress = () => {
+            if (selectionMode === 'shareMap') {
+              // TODO: Navigate to map sharing in subsequent PR
+              console.log('Share map with device:', deviceId);
+            } else {
+              navigation.navigate('SelectInviteeRole', {
+                name: name || '',
+                deviceId: deviceId,
+                deviceType: deviceType,
+              });
+            }
+          };
+
+          return (
+            <TouchableOpacity
+              key={deviceId}
+              disabled={isDisconnected}
+              onPress={handlePress}>
+              <View style={styles.deviceCard}>
+                <DeviceNameWithIcon
+                  style={{flexShrink: 1}}
+                  name={name || ''}
+                  deviceConnectionStatus={status}
+                  deviceType={deviceType}
+                  deviceId={deviceId}
+                  iconSize={75}
+                />
+              </View>
+            </TouchableOpacity>
+          );
+        })}
+      </View>
+    </ScrollView>
+  );
+};
+
+SelectDevice.navTitle = m.title;
+SelectDevice.navTitleMapShare = m.titleMapShare;
+
+type GetSelectableDevicesParams = {
+  peers: PublicPeerInfo[];
+  projectMembers: MemberInfo[];
+  selectionMode: SelectionMode;
+};
+
+export function getSelectableDevices({
+  peers,
+  projectMembers,
+  selectionMode,
+}: GetSelectableDevicesParams): PublicPeerInfo[] {
+  if (selectionMode === 'shareMap') {
+    // For map sharing, return peers that are members of the current project
+    // TODO: possibly update this to include members from all projects the user is part of
+    return peers.filter(device => {
+      const isProjectMember = projectMembers.some(
+        member => member.deviceId === device.deviceId,
+      );
```

Comment:
people who have been blocked or who have left will still return as true in this case. So we need to check that they are not of role `blocked` or `left` here.